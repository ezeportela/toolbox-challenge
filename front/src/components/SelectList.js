import React from "react";

import Form from "react-bootstrap/Form";

function SelectList({ items, placeholder }) {
  return (
    <>
      {items.length > 0 && (
        <Form.Select aria-label={placeholder}>
          <option>{placeholder}</option>
          {items.map(({ name, value }) => (
            <option key={value} value={value}>
              {name}
            </option>
          ))}
        </Form.Select>
      )}
    </>
  );
}

export default SelectList;
