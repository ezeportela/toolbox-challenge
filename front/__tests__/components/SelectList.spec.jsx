import React from "react";
import renderer from "react-test-renderer";
import SelectList from "../../src/components/SelectList";
import { filesList } from "../mocks/files.mocks";

describe("SelectList (Snapshot)", () => {
  it("Should render component", () => {
    const props = {
      items: filesList.map((value) => ({ name: value, value })),
      placeholder: "select list",
      onChange: (target) => target,
      className: "",
    };
    const component = renderer.create(<SelectList {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
