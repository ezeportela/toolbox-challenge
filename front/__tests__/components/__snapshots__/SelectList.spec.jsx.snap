// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList (Snapshot) Should render component 1`] = `
<select
  aria-label="select list"
  className="form-select"
  onChange={[Function]}
>
  <option
    value=""
  >
    select list
  </option>
  <option
    value="file.csv"
  >
    file.csv
  </option>
  <option
    value="test.csv"
  >
    test.csv
  </option>
</select>
`;
